packages/front-end/components/Settings/SharedConnectionSettings.tsx

```diff
@@ -0,0 +1,44 @@
+import { DataSourceSettings } from "@back-end/types/datasource";
+import { ChangeEventHandler } from "react";
+import Tooltip from "@/components/Tooltip/Tooltip";
+import Field from "@/components/Forms/Field";
+
+export interface Props {
+  settings: Partial<DataSourceSettings>;
+  onSettingChange: ChangeEventHandler<HTMLInputElement | HTMLSelectElement>;
+}
+
+export default function SharedConnectionSettings({
+  settings,
+  onSettingChange,
+}: Props) {
+  return (
+    <>
+      <div className="row">
+        <div className="col-md-12">
+          <Field
+            name="maxConcurrentQueries"
+            type="number"
+            label={
+              <>
+                Maximum Concurrent Queries (Optional){" "}
+                <Tooltip
+                  body={
+                    "When executing queries against this datasource, if this many queries are already" +
+                    " running then new connections will wait for existing connections to finish. This" +
+                    " limit is not exact, e.g. if set to 100 it still might allow slightly over 100" +
+                    " queries to run simultaneously due to race conditions"
```

Comment:
Is it worth being more precise here and customer friendly and say something like the following?

```suggestion
                    " queries to run simultaneously if many are initiated by a single experiment update
```

Since that's basically the only update that would spawn more than 1-2 queries now and in the foreseeable future?